packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -82,31 +81,35 @@ export function makeWallet({
   /** @type {Mapping<Brand>} */
   const brandMapping = makeMapping('brand');
   /** @type {Mapping<Contact>} */
-  const contactMapping = makeMapping('contact');
+  const contactMapping = makeMapping(
+    'contact',
+    { useLegacyMap: true }, // because contacts have identity!
```

Comment:
I think contacts are intended to by pass-by-copy records, so we can get rid of `useLegacyMap`, but maybe I'm misunderstanding why the legacy map must be used.

They have one property `depositBoardId` and they are filtered on the basis of this property. I don't see the identity of the object being used anywhere, but I could be wrong. 

@michaelfig, I think you added this part. Can you check my work here?

Reply:
We don't yet have maps that look up based on the contents of pass-by-copy records. That's what the "scalar" qualifier is for. What this used to do, and what it still does with the legacyMap, is use the identity of the record.

If you really do intend to look up by the content, we'll be able to switch away from legacy once we have maps that do that.

Reply:
Serious question: is there a significant use case for record-contents-as-keys?  (Aside from the basic principle that orthogonality means never having to explain which cases work and which don't.)

Reply:
OTOH, IIUC, if you use only `depositBoardId` as a key, then it would be a correct use of a scalarMap and we could retire this legacy use. That's an invasive enough semantic experiment that I'll leave that in both of your hands for after this PR. Resolving.

Reply:
> Serious question: is there a significant use case for record-contents-as-keys? (Aside from the basic principle that orthogonality means never having to explain which cases work and which don't.)

Sorry I noticed this only after merging. This is a central question but I'm breaking for the day. Short answer is: This enables us to implement set-amounts in terms of sets. This needs expansion ;)